### PR TITLE
Converting ROAR Datatable to clientside filter and sort

### PR DESCRIPTION
**Pull request by lucasxsong**
todo:

- [x] pass in non paginated data to component
- [x] figure out why the view support tag now shows a percentile
- [x] move support level color/tagging logic to separate component to improve load speeds
- [x] get extraneous filters to work with current data (filter by grade/school)
- [x] add support for filter by support level
- [x] add loading indicator while tabledata loads in 
- [x] bump score report link logic to separate component
- [x] bump school name logic to separate component
- [x] add filter by optional/assessed parameters
- [x] add filter by unreliable/reliable parameters
- [x] fix bug where filter application for support levels causes all entries to disappear
- [x] convert progress chart to client side filter and sort as well
- [x] refactor scoretabletag to only compute score data for the column it is rendered in
- [x] add progresstabletag to return progress tag
- [x] fix listOrgs datatable not showing up correctly after refactor
- [x] fix bug where calling reset filter does not reset filters

### CODE REVIEW

File: src/components/RoarDataTable.vue
Comment (kellyel) [inline src/components/RoarDataTable.vue:333]: Can I suggest switching from passing a function, and emit an event with the relevant info instead? This data flow keeps consistency with the rest of the app and is Vue's preferred way of handling data flow between parent / child components. I completely understand using this for testing though for debugging purposes! 
This would be easily switched by using `const emits = defineEmits(['resetFilters'])`, `emit('resetFilters')`, where we call RoarDataTable using `@reset-filters="functionName"`

**Review by kellyel — COMMENTED** [review]
Just one comment about code styling, the logic looks good!

File: src/components/RoarDataTable.vue
Comment (lucasxsong) [reply src/components/RoarDataTable.vue:352]: Thanks for the recommendation! I've updated the function call to an emit.

**Review by kellyel — APPROVED** [review]
Approved after merge conflicts are resolved

File: src/components/RoarDataTable.vue
Comment (ksmontville) [inline src/components/RoarDataTable.vue:462]: I had a hard time formatting this as a suggestion due to deleted lines between code blocks, but I hope you get the idea! I am also assuming that useMultiSelect is a boolean or undefined.
```
const useMultiSelect = _get(column, 'useMultiSelect'. false)
const inMatchMode = { value: null, matchMode: FilterMatchMode.IN }

const returnMatchMode = useMultiSelect ? inMatchMode : getMatchMode(dataType)

if (useMultiSelect) {
    options[column.field] = getUniqueOptions(column)
}
```

**Review by ksmontville — APPROVED** [review]

File: src/components/RoarDataTable.vue
Comment (lucasxsong) [reply src/components/RoarDataTable.vue:284]: Thank you for catching this! And thank you for the suggestion to implement it below. I've applied those updates to the branch.

Comment (lucasxsong) [reply src/components/RoarDataTable.vue:284]: Thanks for catching that, I've updated the empty message to be centered in the div.

File: src/components/RoarDataTable.vue
Comment (lucasxsong) [reply src/components/RoarDataTable.vue]: Good suggestion. I've refactored the logic to utilize a map instead and it greatly reduced the bulk of the function. Thanks Kyle!

**Pull request merged**
